src/coreComponents/mesh/generators/VTKMeshGenerator.hpp

```diff
@@ -16,128 +16,128 @@
    *
-   * The pointsets of surface are defined in the same way, using the same
-   * property names "attribute" defined in the input mesh. The pointsets will
+   * The node sets of surface are defined in the same way, using the same
+   * property names "attribute" defined in the input mesh. The node sets will
    * hold a name that is just the attribute index. For instance, if a mesh has three
    * surfaces of interest, with triangles and/or quads holding an attribute value
-   * of 1, 2 or 3, three pointsets named
-   * "1", "2" and "3" will be instantiated by this method
+   * of 1, 2 or 3, three node sets named "1", "2" and "3" will be instantiated by this method
    */
   virtual void generateMesh( DomainPartition & domain ) override;
 
-  virtual void importFields( DomainPartition & elementName ) const override;
+  virtual void importFields( DomainPartition & domain ) const override;
 
-  /**
-   * @brief Create a new geometric object (box, plane, etc) as a child of this group.
-   * @param childKey the catalog key of the new geometric object to create
-   * @param childName the name of the new geometric object in the repository
-   * @return the group child
-   */
-  Group * createChild( string const & childKey, string const & childName ) override;
+  virtual void freeResources() override;
+
+  /// Type of map used to store cell lists
+  using CellMapType = std::map< ElementType, std::unordered_map< int, std::vector< vtkIdType > > >;
 
 private:
+
+  void buildCellBlocks( CellBlockManager & cellBlockManager ) const;
+
+  void buildSurfaces( CellBlockManager & cellBlockManager ) const;
+
+  void importFieldOnCellElementSubRegion( int const regionId,
+                                          ElementType const elemType,
+                                          std::vector< vtkIdType > const & cellIds,
+                                          ElementRegionManager & elemManager,
+                                          arrayView1d< string const > const & fieldNames,
+                                          std::vector< vtkDataArray * > const & srcArrays ) const;
+
+  ///@cond DO_NOT_DOCUMENT
+  struct viewKeyStruct
+  {
+    constexpr static char const * regionAttributeString() { return "regionAttribute"; }
+  };
+  /// @endcond
+
   /**
    * @brief The VTK mesh to be imported into GEOSX.
-   * @note We keep this smart pointer as an attribute because @p importFields uses raw pointers linked to @p m_vtkMesh.
+   * @note We keep this smart pointer as a member for use in @p importFields().
    */
   vtkSmartPointer< vtkUnstructuredGrid > m_vtkMesh;
 
-  /// Path to the mesh file
-  Path m_filePath;
-
-  std::map< int, std::vector< vtkIdType > > m_regionsHex;
-  std::map< int, std::vector< vtkIdType > > m_regionsTetra;
-  std::map< int, std::vector< vtkIdType > > m_regionsWedges;
-  std::map< int, std::vector< vtkIdType > > m_regionsPyramids;
+  /// Name of VTK dataset attribute used to mark regions
+  string m_attributeName;
```

Comment:
Would it be good to rely on the generic field name binding from `ExternalMeshGenerator` to get this information?

Reply:
This one has a very special meaning and treatment from the other fields. It doesn't map to any GEOSX field and gets very special treatment. I think I would prefer to keep it separate.